### ISSUE TITLE
Azure AD Edit Config always picks Standard endpoint in radio button

### ISSUE THREAD

**Issue by maxsokolovsky** [rancher/dashboard#6428]
**Setup**
- Rancher version: v2.6-head near v2.6.7 release
- Browser type & version: Firefox 102

**Describe the bug**
This is relevant for existing setups of Azure AD. The Edit Config screen always selects the Standard endpoint as a value in the radio button, even if the set up has China as the endpoint, or Custom.

**To Reproduce**
This does not happen immediately after registering Azure AD - if you register and then navigate to the Edit Config screen, then everything works correctly, China is selected, if I had registered with a China endpoint and credentials.

But if I navigate away and come back, the issue appears.
1. Register Azure AD with a China endpoint.
2. Go to the local cluster, then back to the Edit Config screen in Auth Providers.

**Result**
The radio button selects the Standard value by default

**Expected Result**
The radio button should select China as the endpoint, since this is how I had configured Azure AD initially.

**Screenshots**

**Additional context**

**Detailed Description**
The same will happen with Custom endpoint configurations. I wonder if the UI could figure out which value to select by querying the auth config and inspecting the value of Graph Endpoint. We know which Standard and China endpoints are valid. If it's none of those, then assume it's Custom.

**Context**
This is not urgent for the v2.6.7 Azure AD changes, just a UI enhancement.

**Comment by anupama2501** [comment]
Verified on v2.6-head 79a5328 upgraded from v2.6.8
Upgrade:
1. Installed rancher server and enabled Azure AD with the `standard` end points.
2. Created 1 rke1 and 1 rke2 node driver all roles Linode clusters
3. As an admin added standard users from Azure AD users in both of the downstream clusters
4. Upgraded rancher server to v2.6-head
5. Edited the config and verified the end points selected were standard endpoints, updated the endpoints to custom end points and added in all the details
6. Verified by editing the config again and noticed the endpoints selected were custom endpoints.

**Comment by anupama2501** [comment]
Re opening the issue on an upgrade use case
Rancher installed on v2.6.6, v2.5.15 upgraded to v2.6-head 79a5328

1. Install rancher on either v2.5.15 or v2.6.6
2. Enable Azure AD auth with standard end points.
3. Upgrade rancher server to v2.6-head
4. Banner to update the endpoints is seen on the UI
5. Before updating the endpoints Click on edit the config >> Notice the standard endpoints are changed to custom endpoints  --> This should have been standard end points and not custom endpoints.
6. Update the endpoints >> edit the config >> still notice the endpoints selected as custom and not standard endpoints

**Comment by anupama2501** [comment]
Verified on v2.6-head 

- Install rancher on either v2.6.5
- Enable Azure AD auth with standard end points.
- Upgrade rancher server to v2.6-head
- Banner to update the endpoints is seen on the UI
- Before updating the endpoints Click on edit the config >> Notice the standard endpoints are set to standard endpoints only 



Re opening for this use case. Above use case works fine
Custom end points:
- Install rancher on either v2.6.5
- Enable Azure AD auth with custom end points.
- Upgrade rancher server to v2.6-head
- Banner to update the endpoints is seen on the UI
- Update the endpoints >> edit the config >> still notice the endpoints selected as custom and not standard endpoints

**Comment by anupama2501** [comment]
Tried this on an upgrade from v2.6.5 >> v2.6-head `2e00266` 

- Install rancher on either v2.6.5
- Enable Azure AD auth with standard end points.
- Upgrade rancher server to v2.6-head
- Banner to update the endpoints is seen on the UI
- Edit the config >> still notice the endpoints selected as custom and not standard endpoints

**Comment by anupama2501** [comment]
Verified on an upgrade from v2.6.5 >> v2.6-head c0ba0f0

- Install rancher on v2.6.5
- Enable Azure AD auth with standard end points.
- Upgrade rancher server to v2.6-head
- Banner to update the endpoints is seen on the UI
- Edit the config >> notice the endpoints selected as standard endpoints

Verified on an upgrade from v2.6.8 >> v2.6-head c0ba0f0
- Install rancher on v2.6.8
- Enable Azure AD auth with standard end points.
- Upgrade rancher server to v2.6-head
- Banner to update the endpoints is seen on the UI
- Edit the config >> notice the endpoints selected as standard endpoints

**Comment by anupama2501** [comment]
Re opening on an upgrade from v2.6.5 >> v2.6-head 3952135

- Install rancher on v2.6.5
- Enable Azure AD auth with custom end points.
- Upgrade rancher server to v2.6-head
- Banner to update the endpoints is seen on the UI
- Before updating the endpoints Click on edit the config >> Notice the standard endpoints are changed to standard endpoints --> This should have been custom end points and not standard endpoints.
- Update the endpoints >> edit the config >> notice the endpoints selected are standard and not custom endpoints

**Comment by maxsokolovsky** [comment]
Tested it today and saw that the fix works but only on initial fresh setup that uses the new endpoints.

If I have a deprecated setup and see the banner, then when I click Edit Config, I see Custom endpoints selected, even though I had set it up with China. When I click Update Endpoints and then Edit Config, the same happens.

Also, it looks like the radio button remembers the last chosen value. For example, if I choose China, then **cancel** and go back, China would be selected. The same is true for all other values.

I am actually not sure how easy this is to fix, not familiar with the UI's inner workings, but I imagine it might be more complicated, since the value of that radio button is not something that the backend stores, so the UI needs to figure it out based on other values on the auth config, which may or may not be trivial.

**Comment by jordojordo** [comment]
The UI determines which radio option is selected by a few factors. There is a [mapping of endpoints](https://github.com/rancher/dashboard/blob/bae50a2c533661a4d1c9b0f4ba938b86388d5209/shell/edit/auth/azuread.vue#L19-L52) that will be iterated over [here](https://github.com/rancher/dashboard/blob/bae50a2c533661a4d1c9b0f4ba938b86388d5209/shell/edit/auth/azuread.vue#L180-L205). If the value of the auth config equals one of the sets of endpoints in the mapping then that option will be selected, otherwise it will be considered custom.

One of the main issues I found while testing was that we do not have a set of credentials that is considered custom (i.e. doesn't match one of the `standard` or `china` options).
I was able to spoof this by running the frontend locally with a deprecated backend and changing one of the `OLD_ENDPOINT` strings to something arbitrary, for instance: `tokenEndpoint: 'https://custom.domain.com'`. This would then show the Custom option being selected. 

Unless there is another way of determining the type of option from an existing config, I don't believe this would be possible without work from the backend.
